### PR TITLE
Nested api field

### PR DESCRIPTION
**Pull request by pmbohm**

### CODE REVIEW

File: ui/src/components/data-entities/form-reducer.js [hunk 1]
```diff
@@ -33,6 +34,21 @@ const formSlice = createSlice({
...
+      state.editItem = {...state.editItem, ...resp};
+    },
+    selectedItemsLoaded: (state, action) => {
+      let resp = {};
+      const key = Object.keys(action.payload._embedded)[0];
+      resp[key] = action.payload._embedded;
+      resp[pluralize.singular(key) + "Selected"] = action.payload.selected;
+      resp[pluralize.singular(key)] = action.payload.selected._links.self.href;
+      state.editItem = {...state.editItem, ...resp};
+    },
```
Comment (utas-tajona) [inline ui/src/components/data-entities/form-reducer.js:51]: I have pull the branch locally and it's crashing here. I don't see where action.payload.selected get fill ?

Especially in the case of new element.

File: ui/src/components/data-entities/customWidgetFields/NestedApiField.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import {Typography} from "@material-ui/core";
+import {selectedItemsRequested, setNestedField} from "../middleware/entities";
+import {markupProjectionQuery} from "../../utils/helpers";
+
+
+const NestedApiField = (props) => {
+
+  let editItemValues = useSelector(state => state.form.editItem);
+  const dispatch = useDispatch();
+
```
Comment (utas-tajona) [inline ui/src/components/data-entities/customWidgetFields/NestedApiField.js:17]: the props has the currently being edited

File: ui/src/components/data-entities/customTooltip.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            </div>
+        );
+      }
+    }
+
+    return (<div className={"custom-tooltip ag-tooltip-hiding"} ></div>)
+
+
+  }
+}
```
Comment (utas-tajona) [inline ui/src/components/data-entities/customTooltip.js:37]: I m not sure we need to add a custom tooltip. ag-grid knows how to deal with large text: https://blog.ag-grid.com/wrapping-column-header-text/

File: ui/src/components/data-entities/customWidgetFields/NestedApiField.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  const entity = props.name ;
+  const pluralEntity = pluralize(entity);
+  //const entities =  pluralEntity.charAt(0).toLowerCase() + pluralEntity.slice(1);
+
+  let itemsList = (editItemValues[pluralEntity]) ?editItemValues[pluralEntity][pluralEntity] : [];
+  let selectedItems = (editItemValues[entity + "Selected"]) ? [editItemValues[entity + "Selected"]].filter(Boolean) : [];
+
```
Comment (utas-tajona) [inline ui/src/components/data-entities/customWidgetFields/NestedApiField.js:24]: this need to be verified if we need to deal with multiple choice. IMO we don't

File: ui/src/components/data-entities/customWidgetFields/NestedApiField.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+  useEffect(() => {
+    let urls = [pluralEntity + "?projection=selection"];
```
Comment (utas-tajona) [inline ui/src/components/data-entities/customWidgetFields/NestedApiField.js:27]: this should be defined in the axios api

**Review by utas-tajona — CHANGES_REQUESTED** [review]

File: ui/src/components/data-entities/customTooltip.js (hunk 3 above)
Comment (pmbohm) [reply ui/src/components/data-entities/customTooltip.js:37]: I dont want to effect heights here

File: ui/src/components/data-entities/customWidgetFields/NestedApiField.js (hunk 4 above)
Comment (pmbohm) [reply ui/src/components/data-entities/customWidgetFields/NestedApiField.js:24]: The code can handle at no cost

File: ui/src/components/data-entities/customWidgetFields/NestedApiField.js (hunk 5 above)
Comment (pmbohm) [reply ui/src/components/data-entities/customWidgetFields/NestedApiField.js:27]: ok yes

Comment (pmbohm) [reply ui/src/components/data-entities/customWidgetFields/NestedApiField.js:27]: moved to helpers.js

File: ui/src/ag-grid.scss [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+.custom-tooltip .body {
+  margin-left: 10px;
+  padding: 5px;
+  white-space: nowrap;
+  border: 2px solid grey;
+  background-color: white;
+  border-radius: 4px;
+}
```
Comment (utas-tajona) [inline ui/src/ag-grid.scss:56]: ag grid should follow the style of material . you can move the the tooltip style within its component

Comment (pmbohm) [reply ui/src/ag-grid.scss:56]: This is the right place for rules applying to AgGrid 
The rules are for Aggrid classes
Here's an example in their documentation: https://www.ag-grid.com/javascript-charts-tooltips/#styling-the-default-tooltip

**Comment by pmbohm** [comment]
@utas-tajona Bug found and squashed

**Pull request merged**